pkg/virtctl/root.go

```diff
@@ -55,10 +59,98 @@ func NewVirtctlCommand() *cobra.Command {
 	return rootCmd
 }
 
-func Execute() {
+func Execute(args []string) error {
 	log.InitializeLogging("virtctl")
-	if err := NewVirtctlCommand().Execute(); err != nil {
-		fmt.Println(strings.TrimSpace(err.Error()))
-		os.Exit(1)
+	// check whether there is a MIME config file
+	// if yes, then set up the args for commands and execute
+	// otherwise normal operation
+	confArgs, _ := parseMime(args)
```

Comment:
Please use `parseMime` also in `NewVirtctlCommand` in the functional tests and add functional tests.

Reply:
There are error cases, like having an invalid file present which are valid errors. Don't ignore them.

Reply:
I covered those tests in unit tests https://github.com/kubevirt/kubevirt/pull/1895/files#diff-2a0b79d92584e47990463ad0767f21ba

I designed the code so it can be unit tested and does not bulk up the functional tests.

Would this be enough?

Reply:
@petrkotas using the code in the functional tests should be as simple as replacing these two lines her: https://github.com/kubevirt/kubevirt/blob/master/tests/utils.go#L2188-L2189 with `virtctl.Execute(append(commandline, args...)`.

In order to tests it then, you can simply run another virtctl command in the vnc test which we already have via a temporary mimetype file.

Reply:
and of course returning an execute function instead of executing it. Something like this:

```go
return func() error {
    return virtctl.Execute(append(commandline, args...)
}
```